scripts/verify_dataset.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+# coding: utf8
+'''Compute various data integrity checks for the OpenMIC dataset.
+
+Example
+-------
+$ ./scripts/verify_dataset.py path/to/openmic path/to/checksums
+'''
+
+import argparse
+import datetime
+import glob
+from joblib import Parallel, delayed
+import json
+import numpy as np
+import os
+import pandas as pd
+import soundfile as sf
+import sys
+import warnings
+
+import openmic.util
+
+
+def hash_one(fname):
+    fkey = openmic.util.filebase(fname)
+    fhash = openmic.util.md5_file(fname)
+    return (fkey, fhash)
+
+
+def hash_collection(filepaths, n_jobs, verbose):
+    dfx = delayed(hash_one)
+    pool = Parallel(n_jobs=n_jobs, verbose=verbose)
+
+    return pd.DataFrame.from_dict(dict(pool(dfx(fn) for fn in filepaths)),
+                                  columns=['md5'], orient='index')
+
+
+def check_md5(filepaths, checksum_file, n_jobs, verbose):
+    df = hash_collection(filepaths, n_jobs, verbose)
+
+    exp = pd.read_csv(checksum_file, index_col=0)
+    df = df.join(exp, how='outer', rsuffix='_expected')
+    if len(df) != len(filepaths):
+        raise warnings.warn('File key mismatch! {} total rows, expected {}'
+                            .format(len(df), len(filepaths)))
+
+    success = (df.md5 == df.md5_expected).all()
+    if not success:
+        percent = 100 * (df.md5 != df.md5_expected).mean()
+        raise warnings.warn('MD5 hash mismatch on {:0.2f}% of records'
+                            .format(percent))
+
+    return success
+
+
+def _check_duration(fname, expected_duration, tolerance):
+    dur = sf.info(fname).duration
+    success = np.abs(dur - expected_duration) < tolerance
+    if not success:
+        raise warnings.warn('{} does not have the expected duration: {} !~ {}'
+                            .format(fname, dur, expected_duration))
+    return success
+
+
+def check_durations(filepaths, expected_duration, n_jobs, verbose, tolerance=0.01):
+    dfx = delayed(_check_duration)
+    pool = Parallel(n_jobs=n_jobs, verbose=verbose)
+    return all(pool(dfx(fn, expected_duration, tolerance) for fn in filepaths))
+
+
+def _check_shape(json_file, expected_shapes):
+    with open(json_file, 'r') as fp:
+        data = json.load(fp)
+
+    success = True
+    for key, shape in expected_shapes.items():
+        act_shape = np.array(data[key]).shape
+        if act_shape != shape:
+            raise warnings.warn('{}:{} has mismatched shapes: {} != {}'
+                                .format(json_file, key, act_shape, shape))
+            success &= False
```

Comment:
could just be `success = False`

Reply:
👍 ya old habits die hard for not re-initing a var